gitlab/v4/objects/groups.py

```diff
@@ -93,12 +93,36 @@ def transfer_project(self, project_id: int, **kwargs: Any) -> None:
         path = f"/groups/{self.id}/projects/{project_id}"
         self.manager.gitlab.http_post(path, **kwargs)
 
+    @cli.register_custom_action("Group", tuple(), ("group_id",))
+    @exc.on_http_error(exc.GitlabTransferProjectError)
+    def transfer_group(self, group_id: Optional[int] = None, **kwargs: Any) -> None:
+        """Transfer a project to this group.
+
+        Requires GitLab ≥14.6.
+
+        Args:
+            group_id: ID of the new parent group. When not specified,
+                the group to transfer is instead turned into a top-level group.
+            **kwargs: Extra options to send to the server (e.g. sudo)
+
+        Raises:
+            GitlabAuthenticationError: If authentication is not correct
+            GitlabTransferProjectError: If the group could not be transferred
+        """
+        path = f"/groups/{self.id}/transfer"
+
+        def data():
+            if group_id is not None:
+                yield ("group_id", group_id)
+
+        self.manager.gitlab.http_post(path, post_data=dict(data()), **kwargs)
```

Comment:
Does this work? I'm not sure why it is a function.

Also when `group_id` is `None` it returns `None`, but `dict(None)` doesn't work so it would fail on L118.

```suggestion
        post_data = {}
        if group_id is not None:
            post_data['group_id"] = group_id

        self.manager.gitlab.http_post(path, post_data=post_data, **kwargs)
```

Tests to validate this works would be great.

Reply:
It's functional style. When `group_id` is `None`, it returns the empty iterator, not `None`, so works correctly.

I'll use the imperative style.

Reply:
Thanks! Make it easier for our little brains on this side 😊